### PR TITLE
Split tests into a few different files

### PR DESCRIPTION
**Pull request by Wimmerer**
I'm not very happy with these splits, but they're all relatively small files, and have some logic to them.

This will be followed by a complete test re-org at some point. For now we're just aiming to reduce the sizes of each test file.

If there's a better split just let me know and I'll change it!

### CODE REVIEW

**Comment by ViralBShah** [comment]
In the future, it would be better to make branches in this repo than forks since that would then allow others who have commit access to directly add to the branch.

**Comment by ViralBShah** [comment]
One more thing that is useful is to move the `USE_GPL_LIBS` check to the individual solver files, and perhaps put all of them in the same directory - so that they can be added to `testgroups`. That way even those will run in parallel across workers. If the threading test in `runtests.jl` can also go into some other file, then `runtests.jl` will be a clean wrapper that just loads everything in `testgroups`.

File: test/sparsematrix_ops.jl [hunk 1]
```diff
@@ -0,0 +1,445 @@
+# This file is a part of Julia. License is MIT: https://julialang.org/license
+
+module SparseTests
```
Comment (ViralBShah) [inline test/sparsematrix_ops.jl:3]: The module name is same across test files. Perhaps should be different?

**Comment by ViralBShah** [comment]
Figured it's easier to merge and do another PR for the solvers.

**Comment by Wimmerer** [comment]
Oh that's why everyone has their own branches. Good to know!

**Pull request merged**